benchmarks/benchmarks/sparse_csgraph_dijkstra.py

```diff
@@ -55,8 +57,14 @@ class DijkstraDensity(Benchmark):
     param_names = ["n", "density"]
 
     def setup(self, n, density):
-       G = scipy.sparse.rand(n, n, density, format='csr') * 100
-       self.graph = G.astype(np.uint32)
+        rng = np.random.default_rng(42)
+        G = scipy.sparse.random_array(
+            shape=(n, n),
+            density=density,
+            format='csr',
+            rng=rng,
+        ) * 100
+        self.graph = G.astype(np.uint32)
```

Comment:
This could be simplified by using the ``data_sampler`` argument from [random_array](https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.random_array.html), however I haven't figured out how yet. I'll update this later if people are interested.

Reply:
I have no issue with the current approach, using `data_sampler` might be overkill

Reply:
The advantage of using `data_sampler` directly gets pretty small between `n=100` and `n=1000`.
But FYI one way to use data_sampler would be:
```python
        G = scipy.sparse.random_array(
            shape=(n, n),
            density=density,
            format='csr',
            rng=rng,
            data_sampler=lambda size : rng.integers(100, size=size, dtype=np.uint32)
        )
```